examples/plot_wlasso.py

```diff
@@ -5,95 +5,116 @@
 
 This example shows how to perform hyperparameter optimization
 for a weighted Lasso using a held-out validation set.
-
+In particular we compare the weighted Lasso to LassoCV on a toy example
 """
 
 # Authors: Quentin Bertrand <quentin.bertrand@inria.fr>
 #          Quentin Klopfenstein <quentin.klopfenstein@u-bourgogne.fr>
-#
+#          Kenan Sehic
 # License: BSD (3-clause)
 
-import time
-
 import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from numpy.linalg import norm
 
-from sklearn.datasets import make_regression
+from celer import Lasso, LassoCV
 from sklearn.model_selection import train_test_split
+from sklearn.metrics import mean_squared_error
+from sklearn.utils import check_random_state
+from scipy.linalg import toeplitz
 
-from celer import Lasso
-
-from sparse_ho.models import wLasso
+from sparse_ho.models import WeightedLasso
 from sparse_ho.criterion import CV
 from sparse_ho.implicit_forward import ImplicitForward
 from sparse_ho.utils import Monitor
 from sparse_ho.ho import grad_search
-from sparse_ho.datasets import get_data
-
-print(__doc__)
 
-dataset = 'rcv1_train'
-# dataset = 'simu'
-
-if dataset != 'simu':
-    X_train, X_val, X_test, y_train, y_val, y_test = get_data(dataset)
-    X_train = X_train[:, :1000]
-    X_test = X_test[:, :1000]
-else:
-    X, y = make_regression(n_samples=100, n_features=100, noise=1)
-    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33)
-    X_train, X_val, y_train, y_val = train_test_split(
-        X_train, y_train, test_size=0.5)
+##############################################################################
+# Dataset creation
+n_samples = 600
+n_features = 600
+rng = check_random_state(0)
+X = rng.multivariate_normal(
+    size=n_samples, mean=np.zeros(n_features),
+    cov=toeplitz(0.5 ** np.arange(n_features)))
+
+
+# Create true regression coefficients of 5 non-zero values
+w_true = np.zeros(n_features)
+size_supp = 5
+idx = rng.choice(
+    X.shape[0], size_supp, replace=False)
+w_true[idx] = (-1) ** np.arange(size_supp)
+noise = rng.randn(n_samples)
+y = X @ w_true
+y += noise / norm(noise) * 0.5 * norm(y)
+
+# Here we split the dataset (X, y) in 3:
```

Comment:
use #############################" so that it renders nicely as text in your doc example